tests/pact/fqns.repl

```diff
@@ -77,3 +77,66 @@
 
 (expect "selects correct test" (modB.get-test) "hello")
 (commit-tx)
+
+;;
+;; Module redeploy name resolution
+;;
+
+(begin-tx)
+(namespace 'free)
+(module test-mod-redeploy-ref g
+  (defcap g () true)
+
+  (defcap test ()
+    (enforce false "boom"))
+
+  (defun f ()
+    (with-capability (test)
+      1))
+  )
+(commit-tx)
+
+(begin-tx)
+(namespace 'free)
+(module test-mod-redeploy-ref g
+  (defcap g () true)
+  (defcap test ()
+    true)
+  (defun f ()
+    (with-capability (free.test-mod-redeploy-ref.test)
+      1))
+  )
+(expect "Namre resolution should find new reference" 1 (f))
+(commit-tx)
+
+; pact 4.7 behaviour
+
+(begin-tx)
+(namespace 'free)
+(module test-mod-redeploy-ref g
+  (defcap g () true)
+
+  (defcap test ()
+    (enforce false "boom"))
+
+  (defun f ()
+    (with-capability (test)
+      1))
+  )
+(commit-tx)
+
+(begin-tx)
+(env-exec-config ["DisablePact48"])
+(namespace 'free)
+(module test-mod-redeploy-ref g
+  (defcap g () true)
+  (defcap test ()
+    true)
+  (defun f ()
+    (with-capability (free.test-mod-redeploy-ref.test)
+      1))
+  )
+
+(expect-failure "Name resolution should find old reference of prev. deployed module" (f))
+
+(commit-tx)
```

Comment:
So there's 3 things being tested here that are relevant: 

1. The first definition of the module has the previous behavior. 
2. The fqn problem is mitigated by the changes in this pr
3. The flag works as expected in masking the upgrade resulting in (1). 

Just to be completely thorough because this is extremely critical code, I think we just need to prove (1) as well.